### PR TITLE
Updated docum of PushSecret

### LINKED ISSUE #2242: Missing things in the docs that we noticed during kubecon
- [x] Better explain PushSecrets and maybe include a diagram
- [ ] Explain all generator types and how it works
- [ ] Explain sourceref (maybe deprecate secrestoreref)
- [ ] Have a guide for pushsecret
(feel free to add more and create sub-issues)

### PR DESCRIPTION
**Pull request by sebagomez**
Fixes #2242 

## Checklist

- [ ] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [ ] All commits are signed with `git commit --signoff`
- [ ] My changes have reasonable test coverage
- [ ] All tests pass with `make test`
- [ ] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: docs/pictures/diagrams-pushsecret-backup.png
Comment (moolen) [inline docs/pictures/diagrams-pushsecret-backup.png:1]: The `pushsecret-backup` diagram is missing in the `diagrams.drawio` file, it has a couple of visual glitches i'd like to fix :bow:

Comment (sebagomez) [reply docs/pictures/diagrams-pushsecret-backup.png:1]: hey @moolen I want to close this one, what's here is better than nothing.
I don't understand your message... did I miss anything? I there anything you wish I fix before we merge it?

Comment (moolen) [reply docs/pictures/diagrams-pushsecret-backup.png:1]: Hey, the `diagrams-pushsecret-backup.png` has been rendered from the `diagrams.drawio` file i suppose. But the provided `diagrams.drawio` doesn't have a tab for `pushsecret-backup`. There is one `pushsecret-basic` though, so i expected that there would be another one for `-backup`. 
Could it be that you forgot to update the `diagrams.drawio`?

I only see this:
pushsec-basic

Comment (moolen) [reply docs/pictures/diagrams-pushsecret-backup.png:1]: It's fine if it got lost in the process, i can re-create the diagram, just let me know. Then we can merge it :+1:

Comment (sebagomez) [reply docs/pictures/diagrams-pushsecret-backup.png:1]: This is finally here!

**Comment by moolen** [comment]
@sebagomez is this ready? It's still a draft but good to be merged for me!

**Comment by sebagomez** [comment]
sorry, I've been sick... yes, it should be ready, will remove the draft
(sorry I can't join our community call today)
(sorry I'm using a comment on PR to let you know :D )

**Review by moolen — APPROVED** [review]
LGTM, thank you!

**Pull request merged**